crates/holochain/src/conductor/conductor.rs

```diff
@@ -1893,82 +1881,42 @@ mod app_status_impls {
-        pub(crate) async fn join_all_pending_cells(&self) -> Vec<CellId> {
-            // Join the network but ignore errors because the
-            // space retries joining all cells every 5 minutes.
-
-            use holochain_p2p::AgentPubKeyExt;
-
-            let tasks = self
-                .mark_pending_cells_as_joining()
-                .into_iter()
-                .map(|(cell_id, cell)| async move {
+            future::join_all(new_cells.iter().map(|(cell, _)| {
+                let sleuth_id = self.config.sleuth_id();
+                async move {
                     let p2p_agents_db = cell.p2p_agents_db().clone();
+                    let cell_id = cell.id().clone();
                     let kagent = cell_id.agent_pubkey().to_kitsune();
                     let maybe_agent_info = match p2p_agents_db.p2p_get_agent(&kagent).await {
                         Ok(maybe_info) => maybe_info,
                         _ => None,
                     };
                     let maybe_initial_arc = maybe_agent_info.clone().map(|i| i.storage_arc);
-                    let network = cell.holochain_p2p_dna().clone();
+                    let agent_pubkey = cell_id.agent_pubkey().clone();
 
-                    aitia::trace!(&hc_sleuth::Event::AgentJoined { node: self.config.sleuth_id(), agent: cell_id.agent_pubkey().clone() });
-
-                    match tokio::time::timeout(JOIN_NETWORK_TIMEOUT, network.join(cell_id.agent_pubkey().clone(), maybe_agent_info, maybe_initial_arc)).await {
-                        Ok(Err(e)) => {
-                            tracing::error!(error = ?e, cell_id = ?cell_id, "Error while trying to join the network");
-
-                            if Self::is_p2p_join_error_retryable(&e) {
-                                Err((cell_id, CellStatus::PendingJoin(PendingJoinReason::Retry(format!("{e:?}")))))
-                            }
-                            else {
-                                Err((cell_id, CellStatus::PendingJoin(PendingJoinReason::Failed(format!("{e:?}")))))
-                            }
-                        }
-                        Err(_) => {
-                            tracing::error!(cell_id = ?cell_id, "Timed out trying to join the network");
-                            Err((cell_id, CellStatus::PendingJoin(PendingJoinReason::TimedOut)))
-                        }
-                        Ok(Ok(_)) => Ok(cell_id),
+                    if let Ok(_) = cell
+                        .holochain_p2p_dna()
+                        .clone()
+                        .join(agent_pubkey, maybe_agent_info, maybe_initial_arc)
+                        .await
+                    {
+                        aitia::trace!(&hc_sleuth::Event::AgentJoined {
+                            node: sleuth_id,
+                            agent: cell_id.agent_pubkey().clone()
+                        });
+                    } else {
+                        todo!("what to do on error?")
```

Comment:
What I'd like to see this code look like is an error from Kitsune that tells you two things. Whether the agent joined the space, and separately whether initial networking succeeded. If the agent join was successful, maybe we log a message here and move on. Otherwise, if the space join failed, then the cell is probably not in a good state and we shouldn't mark it as running here.

Reply:
The only place that has context for doing anything with the error is kitsune itself. Really this should not return a result or anything at all. Unfortunately as it's still a ghost_actor, there is no way to not return a result, so perhaps it is appropriate that it should panic in this location.

Reply:
I think as long as the NetworkInfo contains the joined state so it can be queried, there's nothing for the conductor to do at the moment.